File: app/src/main/java/com/platdmit/simplecloudmanager/domain/repo/ServerBaseRepo.java
package com.platdmit.simplecloudmanager.domain.repo;

import com.platdmit.simplecloudmanager.domain.models.Server;

import java.util.List;

import androidx.lifecycle.MutableLiveData;
import io.reactivex.rxjava3.core.Observable;
import io.reactivex.rxjava3.core.Single;

public interface ServerBaseRepo {
    Observable<List<Server>> getServers();
    Single<Server> getServer(long id);
    void nextUpdate();
}

File: app/src/main/java/com/platdmit/simplecloudmanager/domain/repo/ServerBaseRepo.kt
package com.platdmit.simplecloudmanager.domain.repo

import com.platdmit.simplecloudmanager.domain.models.Server
import io.reactivex.rxjava3.core.Observable
import io.reactivex.rxjava3.core.Single

interface ServerBaseRepo {
    fun getServers(): Observable<List<Server>>
    fun getServer(id: Long): Single<Server>
    fun nextUpdate()
}